AirSkyBoat/AirSkyBoat
# Updates and fixes for Egg Hunt Egg-stravaganza (Event)

**_I affirm:_**


## Please enter a player-facing description

Implemented Egg Hunt Egg-stravaganza 2007, 2008, 2009 and added ability to use custom words (Loxley)

## What does this pull request do? (Please be technical)

* Implements the Egg Helm for 2007+ including the functionality where the player receives the second letter of the party member selected (If they are wearing Egg Helm)
* Implements the Hatchling Shield to dispense egg related items with a conditional check for ToAU era items
* Implements custom words (BONUS_WORDS) into the settings file to be configured per server
* Implements Egg Buffet assembly and disassembly at the MH Moogle
* Removes an extra decoration that was erroneously added to Bastok Markets

https://ffxiclopedia.fandom.com/wiki/Egg_Hunt_Egg-stravaganza_2007/Guide
https://ffxiclopedia.fandom.com/wiki/Egg_Hunt_Egg-stravaganza_2008/Guide
https://ffxiclopedia.fandom.com/wiki/Egg_Hunt_Egg-stravaganza_2009/Guide
(After 2009 the event remains unchanged until 2013 for Prinseggstarta)

## Steps to test these changes

* Form a party to collect a daily egg, selecting a party member who has an Egg Helm equipped
* Repeat without the player having an Egg Helm equipped
* Trade Egg Buffet to the MH Moogle to receive the components
* Trade back the received components to reassemble the Egg Buffet
* Implement a custom word in the settings file and verify the trade is accepted by event Moogles
* Use Hatchling Shield and verify the correct items are received

## Special Deployment Considerations

Update your server settings file with the desired configuration. Example provided in `settings/default/main.lua`.
**Settings name was changed from EGGHUNT to EGG_HUNT for consistency.**
